portfolio/src/main/webapp/script.js

```diff
@@ -28,103 +28,233 @@ function addRandomGreeting() {
 }
 
 //*********************** LOADING SCREEN ***********************
-function setVisible(selector, bool) {
+
+/**
+ * Sets elements to visible or not.
+ * @param {string} id - The id of the HTML element.
+ * @param {bool} bool - true to hide, false to show
+ */
+function setHidden(selector, bool) {
     const el = document.getElementById(selector);
-    el.classList.toggle('hiddenclass', bool);
+    el.classList.toggle('hidden', bool);
 }
 
-function swapLoaderForContent() {
-    setVisible('header', false);
-    setVisible('content', false);
-    setVisible('loaderid', true);
+/**
+ * Gets rid of the loading screen and displays the page
+ */
+function swapLoaderForContent(){
+    setHidden('header', false);
+    setHidden('content', false);
+    setHidden('loaderid', true);
 }
 
+// env variables
+let listenNext = null;
+let retrievedInput = null;
+let match_index = 0;
+let matchingNodes = null;
+
 //*********************** SEARCH BAR FUNCTIONS ***********************
-/**
-  * Swaps the loader for the content and enables search
-  */
+
 window.onload = function(){
-    swapLoaderForContent();
+    swapLoaderForContent(); // displays page after loaded
+
+    // Next button event listener, always active
+    listenNext = document.getElementById("nextbutton");
+    listenNext.addEventListener("click", searchHandler);
+
+
+    //Runs search on enter
     document.getElementById('searchbar').onkeydown = function(e){
         if(e.keyCode == 13){
-            runSearch(document.getElementById('searchbar')
-                .value.toLocaleLowerCase());
+            // i only want single word alphanumerics to go through
+            retrievedInput = document.getElementById('searchbar').value;
+            const regex = new RegExp(/^[a-z0-9]+$/i);
+            if (regex.test(retrievedInput) === false){
+                shakeSearchBar();
+            } else {
+                search(retrievedInput);
+            }
         }
     };
 }
 
+/**
+ * Handler function for the event listener
+ */
+function searchHandler(){
+    highlightHandler(matchingNodes, retrievedInput, match_index);
+}
+
+/**
+ * Code to shake the search bar and make it glow
+ * Calls a bunch of css transforms
+ */
+function shakeSearchBar(){
+    const el = document.getElementById('searchbar');
+    el.preventDefault;
+    el.classList.remove('searchbarglow');
+    void el.offsetWidth; // allows time for the class to be added'
```

Comment:
You shouldn't rely on this niche implementation of your browser. Instead listen on the onanimationend or ontransitionend event before adding your new class if you're concerned with re-triggering the animation.

Reply:
Tysm for this tip